src/Riok.Mapperly/Descriptors/ObjectFactories/ObjectFactoryBuilder.cs

```diff
@@ -28,6 +28,7 @@ public static ObjectFactoryCollection ExtractObjectFactories(SimpleMappingBuilde
             || methodSymbol.IsPartialDefinition
             || methodSymbol.MethodKind != MethodKind.Ordinary
             || methodSymbol.ReturnsVoid
+            || (methodSymbol.IsStatic is false && ctx.Static)
```

Comment:
Nit: For simple conditions this code base usually doesn't use pattern matching: `(!methodSymbol.IsStatic && ctx.Static)`